internal/dag/builder.go

```diff
@@ -889,11 +865,21 @@ func (b *Builder) processTCPProxy(ir *ingressroutev1.IngressRoute, visited []*in
 	b.setStatus(Status{Object: ir, Status: StatusValid, Description: "valid IngressRoute", Vhost: host})
 }
 
-func conditionPath(c projcontour.Condition) string {
-	if c.Prefix == "" {
+func conditionPath(routeCondition, includeCondition *projcontour.Condition) string {
+	pathPrefix := ""
+
+	if includeCondition != nil {
+		pathPrefix = includeCondition.Prefix
+	}
+	if routeCondition != nil {
+		pathPrefix += routeCondition.Prefix
```

Comment:
suggestion, use path.Join() not += because the former will deal with trailing / properly.